packages/medusa/src/loaders/plugins.ts

```diff
@@ -118,47 +142,91 @@ async function runLoaders(pluginDetails: PluginDetails, container: MedusaContain
   )
 }
 
-function registerMedusaApi(pluginDetails: PluginDetails, container: MedusaContainer): void {
+async function registerMedusaApi(
+  pluginDetails: PluginDetails,
+  container: MedusaContainer
+): Promise<void> {
   registerMedusaMiddleware(pluginDetails, container)
   registerStrategies(pluginDetails, container)
 }
 
-function registerStrategies(pluginDetails: PluginDetails, container: MedusaContainer): void {
-  let module
-  try {
-    const path = `${pluginDetails.resolve}/strategies/tax-calculation`
-    if (existsSync(path)) {
-      module = require(path).default
+function registerStrategies(
```

Comment:
while doing that, it could be a good idea to split the strategies registrations into different methods and then call them all in that method as the main one. wdyt?

Reply:
hmm, I don't love it due to the amount of information that will have to be passed back and forth to those functions, especially in order to maintain which priceSelectionStrategies and taxCalculationStrategies that has already been registered along with the logging required.

I have instead made it a `switch(true)` and spaced it out a bit for more readability

Reply:
it was juste a simplification but let's that for another time for now it works perfectly with the switch.
About the backward/forward I don't see what you mean since you have access to the container which allows you to check what have been registered